### PR TITLE
add "optimise selectors" rule

### PR DESCRIPTION
**Pull request by Ex-iT**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,29 @@ This guide aims to improve the way your team uses [jQuery](http://jquery.com/).
...
 * [About jQuery](#about-jquery)
-
+* [Optimise selectors by avoiding excessive specificity](#optimise-selectors-by-avoiding-excessive-specificity)
 
 ## About jQuery
 
 [jQuery](http://jquery.com/) is a utility library for easy DOM access & manipulation, event handling, Ajax and more. By using jQuery you can write consise and expressive code which works across modern and legacy browsers. jQuery has extensive tests, detailed documentation, a large active community and an ecosystem of plugins.
+
+## Optimise selectors by avoiding excessive specificity
+### Why?
```
Comment (jbmoelker) [inline README.md:81]: Also adds more flexibility to how you structure your HTML (separation of concerns)

File: README.md [hunk 2]
```diff
@@ -16,8 +16,29 @@ This guide aims to improve the way your team uses [jQuery](http://jquery.com/).
...
+When selecting elements you should always try to do it by creating a selector that is exactly specific enough for your case.
+
+### How?
+``` javascript
+// Bad
+var $amount = $('.data table.stats td.amount');
```
Comment (jbmoelker) [inline README.md]: Shall we try to give all examples with data-attributes?

**Comment by jbmoelker** [comment]
Should we rephrase this into "Optimise selectors for performance"?  There's a lot of things you can optimise for, e.g. "flexibility".

Also I think we typically find component root elements (`$('[data-my-component]')`) and then `.find()` descendants within (`$component.find('[data-my-component-item]`) or bind (`.on()`) on those (`$component.on('click', '[data-my-component-button]`, function() { /\* ... */ })`). I'm not sure how we align that with this rule.

**Comment by Ex-iT** [comment]
I've updated the examples to not suggest that using `$component.find(...)` is a bad thing. Maybe we should add a section explaining why we commonly use this approach?

**Comment by reinoute** [comment]
Can we not make it more specific and say you should aim to only have a single selector?
Should we say something about using class selectors, id selectors and data-attributes?

Regarding targeting selectors inside components: this guide is not about module based development, but you could explain that `$('[data-table]').find('[data-table-amount]')` is better than just doing `$('[data-table-amount]')`;

What's our recommendation around using [data()](https://api.jquery.com/data/#data-html5) for reading data-attribute values?

**Comment by jbmoelker** [comment]
@Ex-iT @jpsc @reinoute tackle this one in person tomorrow? I'm not convinced yet :)

And @reinoute regarding "What's our recommendation around using data() for reading data-attribute values?", move that to a separate issue?

**Pull request merged**